### PR TITLE
Version Catalog dependencies with toml

### PR DESCRIPTION
**Pull request by MDikkii**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -11,14 +11,14 @@ buildscript {
...
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.0.0'
-        classpath 'com.google.gms:google-services:4.3.8'
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.5.10"
-        classpath "com.github.ben-manes:gradle-versions-plugin:0.33.0"
-        classpath "io.gitlab.arturbosch.detekt:detekt-gradle-plugin:${versions.detekt}"
-        classpath "com.google.dagger:hilt-android-gradle-plugin:${versions.hilt}"
-        classpath "androidx.navigation:navigation-safe-args-gradle-plugin:2.3.5"
-        classpath "org.jlleitschuh.gradle:ktlint-gradle:10.0.0"
+        classpath(libs.gradle)
```
Comment (krzdabrowski) [inline build.gradle:14]: I didn't know you can handle classpath plugins as libraries, thanks! 🙂

**Review by krzdabrowski — APPROVED** [review]
Personally I would bundle much more dependencies to make `gradle` files more compact but I see your point 🙂

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+### Classpath
+gradle = "7.0.0"
+googleServices = "4.3.10"
+kotlinPluginGradle = "1.5.10"
+gradleVersionsPluginGradle = "0.33.0"
+navigationSafeArgsPluginGradle = "2.3.5"
+ktlintGradle = "10.0.0"
+
+[libraries]
```
Comment (Ant8) [inline gradle/libs.versions.toml:57]: nit - what do you think about sorting the dependencies inside groups, eg. in the first paragraph

```
activity
core
corektx
frag
```

File: gradles/deps.gradle [hunk 3]
```diff
@@ -1,65 +1,27 @@
...
-    implementation "androidx.compose.ui:ui:${versions.compose}"
-    implementation "androidx.compose.ui:ui-tooling:${versions.compose}"
-    implementation "androidx.compose.foundation:foundation:${versions.compose}"
-    implementation "androidx.compose.material:material:${versions.compose}"
-    implementation "androidx.compose.material:material-icons-core:${versions.compose}"
-    implementation "androidx.compose.runtime:runtime-rxjava3:${versions.compose}"
-
-    implementation "androidx.activity:activity-compose:1.3.0"
-    implementation "androidx.lifecycle:lifecycle-viewmodel-compose:1.0.0-alpha07"
+    implementation(libs.kotlinStdLib)
```
Comment (Ant8) [inline gradles/deps.gradle:2]: likewise - sorting :)

File: user/src/main/java/com/example/user/presentation/login/LoginPresenter.kt [hunk 4]
```diff
@@ -8,6 +8,8 @@ import dagger.hilt.android.lifecycle.HiltViewModel
 import io.reactivex.rxjava3.core.Flowable
 import javax.inject.Inject
 
+const val LOGIN_THRESHOLD = 3
```
Comment (Ant8) [inline user/src/main/java/com/example/user/presentation/login/LoginPresenter.kt:11]: private?

**Review by Ant8 — APPROVED** [review]
LGTM

File: user/src/main/java/com/example/user/presentation/login/LoginPresenter.kt (hunk 4 above)
Comment (MDikkii) [reply user/src/main/java/com/example/user/presentation/login/LoginPresenter.kt]: Good catch! Done

File: gradles/deps.gradle (hunk 3 above)
Comment (MDikkii) [reply gradles/deps.gradle]: 👍

File: gradle/libs.versions.toml (hunk 2 above)
Comment (MDikkii) [reply gradle/libs.versions.toml:56]: 👍

**Review by krzdabrowski — APPROVED** [review]

**Pull request merged**